force-app/main/default/classes/FilesUploadService.cls

```diff
@@ -0,0 +1,58 @@
+/*
+ * @author Lavanya Bonthala
+ * @date   13/09/2022
+ * @description Service Class containing methods specific to FilesUploadService
+ */
+public with sharing class FilesUploadService {
+     /*******************************************************************************************************
+      * @description This method upload files specific to object record
+      * @param RecordId and filedata
+      * @return show success message on file upload
+      */
+    @AuraEnabled
+     public static String uploadFiles(String ParentRecId, String filedata) {
+        try {
+            if(String.isNotBlank(FileData)) {
+                List<FileDataWP> files = (List<FileDataWP>) JSON.deserialize(filedata, List<FileDataWP>.class);
+                System.debug(files);
+                
+
+                List<ContentVersion> cvList = new List<ContentVersion>();
+                for(FileDataWP file : files) {
+                    system.debug('inside files:'+file);
+                    ContentVersion conVer = new ContentVersion();
+                    conVer.ContentLocation = 'S'; // S = Salesforce, E = External Files
+                    conVer.VersionData = EncodingUtil.base64Decode(file.fileContent);
+                    conVer.Title = file.fileName;
+                    conVer.PathOnClient = file.fileName;
+                    cvList.add(conVer);
+                }
+                Insert cvList;
+  				system.debug('cvlist:'+cvlist);
+                List<ContentDocumentLink> cdList = new List<ContentDocumentLink>();
+                for(ContentVersion cv : [SELECT ContentDocumentId FROM ContentVersion WHERE Id =: cvList]) {
+                    ContentDocumentLink conDocLink = New ContentDocumentLink();
+                    conDocLink.LinkedEntityId = ParentRecId;
+                    conDocLink.ContentDocumentId = cv.ContentDocumentId;
+                    conDocLink.Visibility = 'AllUsers';
+                    conDocLink.shareType = 'V'; // V = Viewer, C = Collaborator, I = Inferred
+                    cdList.add(conDocLink);
+                }
+                Insert cdList; 
+                system.debug('cdlist:'+cdList);
+            } else {
+                return 'record Id is Blank';
```

Comment:
Can return as a custom exception as record Id cannot be blank. Have the message in custom labels